RELEASES.md

```diff
@@ -1,3 +1,440 @@
+
+These abstractions can be combined in various ways, yielding full control to the users. For example, in the following only `a` and `b` share a common storage cell while `c` lives in its own:
+
+```rust
+pub struct Spreaded {
+    a: i32,
+    b: ink_storage::Lazy<i32>,
+}
+
+#[ink(storage)]
+pub struct Packed {
+    packed: ink_storage::Pack<Spreaded>,
+    c: [u8; 32],
+}
+```
+
+### Spreading Array Cells
+
+If we prefer to store all bytes of `c` into their own storage cell we can make use of the `SmallVec` data structure. The `SmallVec` is a high-level data structure that allows to efficiently organize a fixed number of elements similar to a Rust array. However, unlike a Rust array it acts lazily upon the storage and spreads its elements into different cells.
+
+```rust
+use typenum::U32;
+
+pub struct Spreaded {
+    a: i32,
+    b: ink_storage::Lazy<i32>,
+}
+
+#[ink(storage)]
+pub struct Packed {
+    packed: ink_storage::Pack<Spreaded>,
+    c: SmallVec<u8, U32>,
+}
+```
+
+### Opting-out of Storage
+
+If you are in need of storing some temporary information across method and message boundaries ink! will have your back with the `ink_storage::Memory` abstraction. It allows you to simply opt-out of using the storage for the wrapped entitiy at all and thus is very similar to Solidity's very own `memory` annotation.
+
+An example below:
+
+```rust
+#[ink(storage)]
+pub struct OptedOut {
+    a: i32,
+    b: ink_storage::Lazy<i32>,
+    c: ink_storage::Memory<i32>,
+}
+```
+
+The the above example `a` and `b` are normal storage entities, however, `c` on the other hand side will never load from or store to contract storage and will always be reset to the default value of its `i32` type for every contract call.
+It can be accesses from all ink! messages or methods via `self.c` but will never manipulate the contract storage and thus acts wonderfully as some shared local information.
+
+### Dynamic Storage Allocator
+
+In the previous section we have seen how the default mode of operation is to spread information and how we can opt-in to packing information into single cells via `ink_storage::Packed`.
+
+However, what if we wanted to store a vector of a vector of `i32` for example?
+Naturally a user would try to construct this as follows:
+
+```rust
+use ink_storage::Vec as StorageVec;
+
+#[ink(storage)]
+pub struct Matrix {
+    values: StorageVec<StorageVec<i32>>,
+}
+```
+
+However, this will fail compilation with an error indicating that `StorageVec<T>` requires for its `T` to be packed (`T: PackedLayout`) which `StorageVec<T>` itself does not since it stores all of its elements into different cells always. The same applies to many other storage data sturctures provided by `ink_storage` and is a trade-off the ink! team decided for the case of efficiency of the overall system.
```

Comment:
```suggestion
However, this will fail compilation with an error indicating that `StorageVec<T>` requires for its `T` to be packed (`T: PackedLayout`) which `StorageVec<T>` itself does not since it always stores all of its elements into different cells. The same applies to many other storage data sturctures provided by `ink_storage` and is a trade-off the ink! team decided for the case of efficiency of the overall system.
```